### DISCUSSION TITLE
Newbie - Updating server side and refreshing view

### DISCUSSION THREAD

**Discussion by scottjo** [livewire/livewire#7201] [1. Help]
Good day folks and please accept my apologies for a newbie question. I have a component with an array as a messageList - during back end processing I want to push messages to that array and the view to 'render' those messages as they are added. I knocked up a small component to try this out. 

    
    class Importer extends Component
    {
        public array $messagesList = [];

        protected $listeners = ['refreshComponent' => '$refresh'];

        public function mount(): void
        {
            $this->messagesList = [];
        }

        public function render(): Application|Factory|View|\Illuminate\Foundation\Application
        {
            return view('livewire.importer');
        }

        public function startImport(): void
        {
             $i = 0;
             while ($i < 10) {
                $this->messagesList[$i] = now()->format('H:m:s');
                $this->dispatch('refreshComponent');
                sleep(1);
                ray($i);
                $i++;
            }
       }

    }
`

and I have a simple view

```
    <div>
         <button id="startImport" wire:click="startImport">Import</button>
         <ul>
             @foreach ($messagesList as $message)
                <li>{{ $message }}</li>
             @endforeach
         </ul>
     </div>
```

Now when I click the 'Import' button - it kicks off and I don't see the UL until the end of loop - no message until it has iterated 10 seconds. Once finished I get all 10 entries appear at once on the screen. Looking at other similar questions on here I've added the listener connected to $refresh and dispatching the message during the loop but has made no difference. What am I doing wrong?

**Comment by joshhanley** [comment]
@scottjo the reason that doesn't work, is because php will finish the request before returning the response to the browser. So all those dispatches will be dispatched all at the same time when Livewire's javascript receives the response and processes them.

You might be able to achieve what you're hoping for with `wire:stream` if you're using Livewire v3. See here for details https://livewire.laravel.com/docs/wire-stream

Hope this helps!

**Reply by scottjo** [reply]
Thanks Josh for the explanation and the suggestion. I'll take a look and report back.
